docs/modules/data-structures/pages/working-with-external-data.adoc

```diff
@@ -250,6 +250,106 @@ include::ROOT:example$/dds/map/PersonEntryStore.java[tag=personms]
 
 TIP: link:https://github.com/hazelcast/hazelcast-code-samples/tree/master/distributed-map/mapstore/src/main/java[See more MapStore/MapLoader code samples].
 
+== Offloading MapStore and MapLoader Operations
+
+To improve cluster throughput, all MapStore operations do not block partition threads.
+Each partition in a member is managed by a single thread, ensuring that the write operations to a given partition are handled
+one at a time in first-in-first-out order. However, MapStore operations may take a long time to connect to the external data source and complete, blocking the affected partitions. 
+To avoid blocking partition threads, MapStore operations that communicate with the external data source are handled asynchronously, using an <<configuring-the-offload-executor, executor>>.
+
+=== Limitations
+
+* The `init()` and `destroy()` methods of the `MapLoaderLifecycleSupport` interface cannot be offloaded. These methods are triggered only once in the whole lifecycle of a map.
+* MapStore methods `loadAll()` and `loadAllKeys()` cannot be offloaded.
+* Offloading cannot be used for xref:tiered-storage:overview.adoc[tiered storage], since you cannot enable both MapStore and tiered storage for a map.
+
+=== Blocking Partition Threads
+
+If you want to block partition threads until MapStore operations are complete, 
+you can configure the `offload` setting for the MapStore. It is enabled by default (`true`) and can be configured for each map in your cluster.
```

Comment:
>If you want to block partition threads until MapStore operations are complete, 
you can configure the `offload` setting for the MapStore. It is enabled by default (`true`) and can be configured for each map in your cluster.

To restore the behavior in previous releases than 5.2, you can set `offload` setting for the `MapStoreConfig` to `false` which is `true` by default. This way partition threads will be blocked during MapStore/MapLoader api calls.

Reply:
We try to avoid mentioning the product release numbers, since the documentation is versioned.
Instead, I can simply give the idea of "set to "false" to block".

Reply:
My point is we should emphasis that the point of making it false is only bringing old behavior back. Giving version number is not necessary.